### PR TITLE
Minor Updates to API Materialized Views

### PR DESCRIPTION
**Pull request by tony-sappe**
## Main Edits: tweaks to a number of materialized views and some existing indexes
- Added columns for ES data loading
- Added extra step to matview SQL which grants read privileges to `readonly` user
- Added `award_description` to keyword search
- Improved Agency Searches (no longer fuzzy searching)
- Change recipient_location_* and pop_* indexes in `summary_transaction_month_view` and `summary_transaction_view` to partial indexes

- [x] Recreate Matviews on Dev
- [x] ES team review on new columns
- [x] Backend Review

### Universal_Transaction_MatView
-  Added col: transaction_id
-  Added col: modification_number
- Added col: awarding_agency_id
- Added col: funding_agency_id
- Readded col: pop_county_name
- Readded col: recipient_location_county_name
- Added index: awarding_agency_id
- Added index: funding_agency_id
- Removed index: gin_awarding_toptier_agency_name
- Removed index: gin_awarding_subtier_agency_name
- Removed index: gin_funding_toptier_agency_name
- Removed index: gin_funding_subtier_agency_name

### Universal_Award_MatView
- Added col: awarding_agency_id
- Added col: funding_agency_id
- Readded col: pop_county_name
- Readded col: recipient_location_county_name
- Added index: awarding_agency_id
- Added index: funding_agency_id

### Summary_Award_View
- Added col: awarding_agency_id
- Added col: funding_agency_id
- Added index: awarding_agency_id
- Added index: funding_agency_id

### Summary_View
- Added col: awarding_agency_id
- Added col: funding_agency_id
- Added index: awarding_agency_id
- Added index: funding_agency_id

### Summary_Transaction_View
- Added col: awarding_agency_id
- Added col: funding_agency_id
- Added index: awarding_agency_id
- Added index: funding_agency_id
- Removed index: psc2

### Summary_Transaction_Month_View
- Added col: awarding_agency_id
- Added col: funding_agency_id
- Added index: awarding_agency_id
- Added index: funding_agency_id

### CODE REVIEW

File: usaspending_api/awards/v2/filters/matview_filters.py [hunk 1]
```diff
@@ -74,54 +74,54 @@ def matview_search_filter(filters, model):
...
+            awarding_toptier = Q()
+            awarding_subtier = Q()
             for v in value:
                 type = v["type"]
                 tier = v["tier"]
                 name = v["name"]
                 if type == "funding":
                     if tier == "toptier":
-                        funding_toptier.append(name)
+                        funding_toptier |= Q(toptier_agency_id__name=name)
```
Comment (nmonga91) [inline usaspending_api/awards/v2/filters/matview_filters.py]: Should this be `funding_toptier_agency_name`? I see the double underscore, but I did not see the relationship setup in corresponding matview model.

Comment (nmonga91) [reply usaspending_api/awards/v2/filters/matview_filters.py]: Similar question for other changes in this file below.

**Comment by nmonga91** [comment]
Backend review is done. Waiting for ES review before approving PR.

**Comment by tony-sappe** [comment]
Readded agency name btree indexes to the matviews which have these columns:

- awarding_toptier_agency_name
- awarding_subtier_agency_name
- funding_toptier_agency_name
- funding_subtier_agency_name

**Pull request merged**